Client/Frontend/Browser/TabManager.swift

```diff
@@ -173,6 +173,11 @@ class TabManager: NSObject {
             return
         }
 
+        // Make sure to wipe the private tabs if the user has the pref turned on
+        if shouldClearPrivateTabs(), !(tab?.isPrivate ?? false) {
+            removeAllPrivateTabsAndNotify(false)
```

Comment:
Logic question: What happens when the second private tab is opened? There doesn't seem to be any check to see if the `previous` tab is private and this `tab` is not.

It's not clear to me how this works (if it does) and the tests don't show it either.

Nit: prefer `(tab?.isPrivate ?? false)` be broken out into a local variable if you're going to add `(previous?.isPrivate ?? false)` as well.